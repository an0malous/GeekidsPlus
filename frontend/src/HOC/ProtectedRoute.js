import React from 'react';
import { Route, Redirect } from 'react-router-dom';

const ProtectedRoute = ({component: Component, loggedIn, role, accessLevel, ...rest }) => (
    <Route
      {...rest}
      render={props => {
        return loggedIn && role >= accessLevel ? (
          <Component {...rest} />
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: props.location }
            }}
          />
        );
       
      }}
    />
  );

  export default ProtectedRoute;