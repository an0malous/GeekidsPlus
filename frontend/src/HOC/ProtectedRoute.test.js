import React from 'react';
import { shallow } from 'enzyme';
import ProtectedRoute from './ProtectedRoute';

it('expect ProtectedRoute component to render', ()=>{
   expect(shallow(<ProtectedRoute />)).toMatchSnapshot();
});