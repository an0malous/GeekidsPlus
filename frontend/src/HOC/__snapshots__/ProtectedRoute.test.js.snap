// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect ProtectedRoute component to render 1`] = `
<Route
  render={[Function]}
/>
`;
